packages/iov-keycontrol/src/keyring-entries/slip10.ts

```diff
@@ -96,14 +115,14 @@ export class Ed25519HdKeyringEntry implements KeyringEntry {
     for (const record of decodedData.identities) {
       const identity: LocalIdentity = {
         pubkey: {
-          algo: Ed25519HdKeyringEntry.algorithmFromString(record.localIdentity.pubkey.algo),
+          algo: Slip10KeyringEntry.algorithmFromString(record.localIdentity.pubkey.algo),
```

Comment:
This seems to allow multiple algorithms in one keyring, even if the above code uses only one curve.... Either way is fine, but if we only do one, I would read algorithm from top level.

Reply:
This is because data is duplicated. Each identity has an algorithm identifier, that can be used when the identity is copied out of the entry. This would allow us to create keyring entries in the future, that can produce identities with different curves. This is not necessarily a great thing but the possibility shows that our abstraction is good.

What we could do here in Slip10KeyringEntry is to verify that entry curve and identity algorithm match via an assertion. This would protect against some super crazy programming errors. However, we already cover that created identities are using the right algorithm.

Reply:
https://github.com/iov-one/iov-core/pull/310/commits/4af9a240da36482515cd80faba44c8666a8bf9f3